src/Microsoft.AspNet.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -0,0 +1,206 @@
+    /// </summary>
+    /// <remarks>Resolves application relative URLs.</remarks>
+    [TargetElement("*", Attributes = "itemid")]
+    [TargetElement("a", Attributes = "href")]
+    [TargetElement("area", Attributes = "href")]
+    [TargetElement("link", Attributes = "href")]
+    [TargetElement("base", Attributes = "href")]
+    [TargetElement("video", Attributes = "poster")]
+    [TargetElement("video", Attributes = "src")]
+    [TargetElement("audio", Attributes = "src")]
+    [TargetElement("embed", Attributes = "src")]
+    [TargetElement("iframe", Attributes = "src")]
+    [TargetElement("img", Attributes = "src")]
+    [TargetElement("script", Attributes = "src")]
+    [TargetElement("source", Attributes = "src")]
+    [TargetElement("track", Attributes = "src")]
+    [TargetElement("input", Attributes = "src")]
+    [TargetElement("input", Attributes = "formaction")]
+    [TargetElement("button", Attributes = "formaction")]
+    [TargetElement("form", Attributes = "action")]
+    [TargetElement("blockquote", Attributes = "cite")]
+    [TargetElement("del", Attributes = "cite")]
+    [TargetElement("ins", Attributes = "cite")]
+    [TargetElement("q", Attributes = "cite")]
+    [TargetElement("menuitem", Attributes = "icon")]
+    [TargetElement("html", Attributes = "manifest")]
+    [TargetElement("object", Attributes = "data")]
+    [TargetElement("object", Attributes = "archive")]
+    [TargetElement("applet", Attributes = "archive")]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class UrlResolutionTagHelper : TagHelper
+    {
+        private static readonly IReadOnlyDictionary<string, IEnumerable<string>> ElementAttributeLookups =
+            new Dictionary<string, IEnumerable<string>>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "a", new[] { "href" } },
+                { "area", new[] { "href" } },
+                { "link", new[] { "href" } },
+                { "base", new[] { "href" } },
+                { "video", new[] { "poster", "src" } },
+                { "audio", new[] { "src" } },
+                { "embed", new[] { "src" } },
+                { "iframe", new[] { "src" } },
+                { "img", new[] { "src" } },
+                { "script", new[] { "src" } },
+                { "source", new[] { "src" } },
+                { "track", new[] { "src" } },
+                { "input", new[] { "src", "formaction" } },
+                { "button", new[] { "formaction" } },
+                { "form", new[] { "action" } },
+                { "blockquote", new[] { "cite" } },
+                { "del", new[] { "cite" } },
+                { "ins", new[] { "cite" } },
+                { "q", new[] { "cite" } },
+                { "menuitem", new[] { "icon" } },
+                { "html", new[] { "manifest" } },
+                { "object", new[] { "data", "archive" } },
+                { "applet", new[] { "archive" } },
```

Comment:
Pity the `[TargetElement]` attributes can't be changed via configuration. Nonetheless should this come from configuration or via an `ITagHelperInitializer`? That would at least allow users to subset the actions of this tag helper without turning it off completely or running another tag helper before it.